### PR TITLE
Site setup: Add task for updating menus

### LINKED ISSUE #40345: Site Setup List (checklist): Add checklist step for updating Menus
Until we have full site editing available, we should help guide customers on how to add and edit a menu item on their site.

Since doing so currently will take folks to the customizer, let's see if we can pair up with Jeff Rozic on another get started video for adding/editing menus.

image

### Notes
* The checklist is backed by a server API. We need to know when a task is done (probably the server detecting an edit made to menus). So this will require server patches. It may be also be worth it to make this step **skippable**. (There should be existing examples of this).
* We may need to think about for whom this task is eligible for. (Double check that we don't unset the "doneness" of site setup for folks that have already run through this list).
* Please link to https://wordpress.com/customize/menus/<site> (Same as Edit Menus in quick links)
* Please add required analytics that we expect to this new task

Additional context: pbAPfg-aF-p2

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

- Add new task to the site setup checklist for updating the menus of the site.
- Task opens a dialog with the Site Menus support article (where a video will be added in the future).
- Add support for a custom button label and URL in the support article dialogs.
- Dialog links to the Customizer's menus section.
- Display skip button as a link.
- Removes the "You completed a task!" notice when skipping a task.

Screen Shot 2020-04-08 at 11 24 13

Screen Shot 2020-04-07 at 14 42 02

#### Testing instructions

- Load Calypso running this branch and switch to a simple site with the checklist incomplete.
- Make sure there is a task for updating the menus (it should be the 2nd one if the site title has not been set yet).
- Make sure clicking on "Start with a tutorial" opens a dialog with the content of https://wordpress.com/support/menus/.
- Go to the Customizer and update the site menu.
- Go back to My Home and make sure the task has been completed.

Fixes #40345

### CODE REVIEW

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 1]
```diff
@@ -986,6 +996,29 @@ class WpcomChecklistComponent extends PureComponent {
...
+
+	renderSiteMenuUpdateTask = ( TaskComponent, baseProps, task ) => {
+		const { translate } = this.props;
+
+		return (
+			<TaskComponent
+				{ ...baseProps }
+				bannerImageSrc="/calypso/images/stats/tasks/personalize-your-site.svg"
+				completedButtonText={ translate( 'Edit' ) }
+				completedDescription={ translate( 'You can edit your site menu whenever you like.' ) }
```
Comment (a8ci18n) [inline client/my-sites/checklist/wpcom-checklist/component.jsx:1008]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **16** times:
`translate( 'You can edit your site title whenever you like.' )` *ES Score: 7*

**Comment by danhauk** [comment]
> Display skipped label if user chooses to skip this step (wondering if this could be part of a follow-up).

Yeah, I think this can be part of a follow-up. Showing a skipped state is more of a functionality change for the checklist as a whole.

Now I'm second guessing the wording of the action. Does "Skip" create different expectations? I don't think I'd expect it to be marked complete. Maybe "Mark as complete" would be better since that's effectively what we're doing here? cc @Automattic/dotcom-manage-design

**Comment by gwwar** [comment]
> Display skip button as a link (wondering if this could be part of a follow-up).

I think that'd be fine, as this would also modify other checklists (Jetpack).

> Display skipped label if user chooses to skip this step (wondering if this could be part of a follow-up).

I think this is an optional enhancement for later. Since I don't think we store enough state for this atm, and we have other existing tasks that may be skipped.

**Comment by kwight** [comment]
> Now I'm second guessing the wording of the action. Does "Skip" create different expectations? I don't think I'd expect it to be marked complete.

I'm wondering this too. I also got it in my head after closing the modal as, "skip the tutorial, go right to the menu editing".

If we click Skip, we immediately get a "you completed a step!" notice, and the menu step under Completed also says you changed your menu.

Screen Shot 2020-04-07 at 12 05 42 PM

I don't think any other steps have a Skip option yet – can we remove the Skip action until we sort out a better skip experience?

**Comment by gwwar** [comment]
@kwight I think it'd be fair to disable the global notice for skipped tasks. Would you like to add that to this PR?

We already have skip tasks for things like the app download.

**Review by gwwar — APPROVED** [review]
This tests well for me @mmtr! Nice touch with the menu opening the support page. Some follow up steps:

- [x] When skipping tasks we can remove the global notice (here or in a follow up)
- [x] I think we're missing a tracks event `calypso_checklist_task_start` for this task (it fires for the other events) 
- Displaying skip status is fine to leave for a future enhancement

File: client/state/selectors/get-inline-support-article-action-label.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * @param {object} state Global app state
+ * @returns {object} ...
+ */
+export default state => state?.inlineSupportArticle?.actionLabel ?? translate( 'Visit article' );
```
Comment (a8ci18n) [inline client/state/selectors/get-inline-support-article-action-label.js:10]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Visit Article' )` *ES Score: 11*

**Comment by mmtr** [comment]
Thanks everyone for the reviews! Pushed a bunch of commits improving the overall experience:
- Skip buttons are displayed now as links.
- We no longer show the "You completed a task!" notice when skipping a task.
- Added tracking for `calypso_checklist_task_start` events.
- Replace "Edit" with "View tutorial" as button label when the task is completed.

**Pending task**
Screen Shot 2020-04-08 at 11 17 50

**Completed task**
Screen Shot 2020-04-08 at 11 18 06

A confidence check before landing this. Should we wait for the video to be present in the support article before landing this? I think the tutorial could be a bit overwhelming for unexperienced users without a video and might result in more users not being able to complete the checklist.

**Review by kwight — APPROVED** [review]
Great, updates still all work as expected 👍

**Comment by kwight** [comment]
> Should we wait for the video to be present in the support article before landing this?

I don't think we should wait; this step without the video is still much better guidance for menus than nothing at all. And once the video is up, we won't need to do anything oh the Home side anyways – I say we push it out.

**Comment by gwwar** [comment]
yup the video is  💯 a nice to have on the support page, not a blocker

**Pull request merged**